### DISCUSSION TITLE
Support for idle or inactive session timeouts

### DISCUSSION THREAD

**Discussion by arieltorti** [ory/kratos#3172] [Q&A] [answered]
Does Ory Kratos support idle sessions? In the configuration, there's a key for session lifespan but I would like to invalidate sessions faster if they haven't had any activity.

For clarification, when I say idle timeout I'm referring to the amount of time that a session can be inactive (i.e: not communicating with the server) before the server expires the session.

If the feature is not natively provided by Kratos, is there a workaround available? Maybe intercept the calls to `/sessions/whoami` and invalidate the session if the idle timeout was triggered?

**Comment by vinckr** [comment, marked as answer]
Hello @arieltorti 
It is not available out of the box right now, but you have two APIs you can use:

- https://www.ory.sh/docs/reference/api#tag/identity/operation/deleteIdentitySessions
- https://www.ory.sh/docs/reference/api#tag/identity/operation/disableSession

Depends how your setup is you can use those to invalidate all sessions of an identity or just a specific session.

Let me know if that helps.
